### ISSUE TITLE
In the rectangle-selecting UI, give ability to "unite" multiple rectangular regions.

### ISSUE THREAD

**Issue by be5invis** [Snipaste/feedback#458]
Snipaste has a rectangle-selecting mode, that you can directly click to decide a snapshot area. This feature is convenient, however given that many windows is separated into small pieces, when wanting to take a large window/region, it becomes inconvenient.
Could snipaste include a "unite" mode (like -- holding some key) to combine multiple small rectangles, using their bounding box, to get a large window conveniently?

**Comment by liulex** [comment]
Good idea, and we may not restrict to using the bounding box (like filling the gaps with transparent pixels).

However, it would require huge changes to my current code. So... stay tuned.

**Comment by deltaidea** [comment]
Suggestion: after making a selection, make dragging selection edges snap to UI elements.

If I want a bounding box around two UI elements, I will select the first one and then drag a corner to expand around the second UI element. If you make it softly snap (let's say, 5px around UI element edges snaps to them) it would be easy to be precise.

This sounds more intuitive to use and might be easier to implement.

**Comment by Geeyun-JY3** [comment]
> Suggestion: after making a selection, make dragging selection edges snap to UI elements.
> 
> If I want a bounding box around two UI elements, I will select the first one and then drag a corner to expand around the second UI element. If you make it softly snap (let's say, 5px around UI element edges snaps to them) it would be easy to be precise.
> 
> This sounds more intuitive to use and might be easier to implement.

@deltaidea I think #1120 is what you want.
